syntax = "proto3";

package auth;

service Auth {
  rpc SignUp (SignUpRequest) returns (AuthResponse);
  rpc Login (LoginRequest) returns (AuthResponse);
  rpc ValidateToken (ValidateRequest) returns (ValidateResponse);
}

message SignUpRequest { string name = 1; string email = 2; string password = 3; }
message LoginRequest { string email = 1; string password = 2; }

message AuthResponse { string user_id = 1; string access_token = 2; }

message ValidateRequest { string token = 1; }
message ValidateResponse { bool valid = 1; string user_id = 2; string claims_json = 3; }





